packages/camera/camera_android_camerax/android/src/main/java/io/flutter/plugins/camerax/CameraControlHostApiImpl.java

```diff
@@ -83,6 +83,12 @@ public void onSuccess(Void voidResult) {
             }
 
             public void onFailure(Throwable t) {
+              if (t instanceof CameraControl.OperationCanceledException) {
+                // Operation was canceled due to camera being closed or a new request was submitted, which
+                // is not actionable and should not block a new value from potentially being submitted.
+                result.success(null);
+                return;
+              }
```

Comment:
A semi-related fix: some `CameraControl` operations will throw an exception if interrupted or could not have completed due to the camera being closed. Changed this to result in a success since this is not an exception could/should be handled by someone using the plugin.

For an example of why, the example camera app has a slider to change the exposure offset, and this results in many of these exceptions being thrown because it makes many rapid calls to `CameraControl.setExposureCompensationIndex`.

Reply:
I believe the camera closes if there is an error or if all `UseCase`s are bound. Regardless, I agree that it should not silently fail; that's why I made it return `null` and added a `CameraErrorEvent` to the error event stream on the Dart side. Open to discussion on that choice though :) Ideally, we could know when the camera closes versus the operation being canceled because a new one was set, because I don't think we should be surfacing the latter as an exception as it does without this `catch` statement.

Reply:
Makes sense! I made the comment after seeing `result.success` here but didn't connect that it led to an error in `camera_control.dart`, thanks for clarifying